### PR TITLE
nutdrv_qx: add support for SNR-UPS-LID-XXXX

### PR DESCRIPTION
**Pull request by askazancev**
Hello everyone. 
In this PR I suggest adding support for UPSes series SNR-UPS-LID-XXXX to nutdrv_qx as a USB device.
An end-user should reference explicitly the `snr` subdriver in `ups.conf` settings because of using the same values of VendorID/ProductID as fabula_subdriver, fuji_subdriver, and krauler_subdriver.

### CODE REVIEW

**Review by jimklimov — CHANGES_REQUESTED** [review]
Looks quite solid to me, thanks! Got one comment for the code to fix, but otherwise looks ok to merge if nobody vetoes this in the next few days :)

Hope @zykh can take a look at this regarding Qx driver/protocol evolution in general.

Just in case, these devices do not need a separate subdriver file for some more complex logic or lookup tables as in e.g. https://github.com/networkupstools/nut/pull/638/files#diff-a94c7d9bf3942c23f12a818a5f0d6f75c8cfb259488ea3c1b93991c0e09b7ca6 ?

File: drivers/nutdrv_qx.c
Comment (askazancev) [reply drivers/nutdrv_qx.c:1042]: Thank you for taking a look at it!

Do you mean to add something like that before the first cycle ([line 1029](https://github.com/networkupstools/nut/pull/1008/files/b1b39d9b642a16696a831eec1699858439c7a866#diff-cb890ac7b82cb7a4f861284bcf39cc4f168312b61cb9455eb2755e345c0aa627R1029))?
```
	if (buflen < 102) {
		upsdebugx(4, "size of buf less than 102 byte!");
		return 0;
	}
```
I think it will protect against overflows if the buffer is too small.

**Review by jimklimov — APPROVED** [review]
Thanks for the added check, looks safer now :)

So, waiting a few days if someone else from the community has any concerns....

**Comment by jimklimov** [comment]
One more point came up: the limitation of 102 bytes seemed reminiscent of something - the "hunnox" subdriver introduced with #638 seems to have spell it too at https://github.com/networkupstools/nut/pull/638/files/3a5a758dbbfbb8bdf22a45cdde06fe10d4e0249b#diff-cb890ac7b82cb7a4f861284bcf39cc4f168312b61cb9455eb2755e345c0aa627R922 although the list of commands on top of the (originally fabula) method differs from one introduced for "snr" here.

**Comment by jimklimov** [comment]
Note: nutdrv_qx.c DRIVER_VERSION not bumped here since it was updated a minute ago with merge of PR #638.

File: docs/man/nutdrv_qx.txt
Comment (askazancev) [inline docs/man/nutdrv_qx.txt:327]: Double *sgs*

**Comment by jimklimov** [comment]
Thanks, sorry, copy-paste error :(

File: docs/man/nutdrv_qx.txt
Comment (jimklimov) [reply docs/man/nutdrv_qx.txt:327]: Fixed in master branch after merge, thanks for bringing the attention to this and sorry for the typo.

**Pull request merged**
